tests/test_matrix.py

```diff
@@ -0,0 +1,271 @@
+import pytest
+
+from matrix.matrix import Matrix
+
+CONSTANT_NUMBER_LIST = [50, -2.5, 0, 2]
+LEGAL_MATRIX_LIST = [(()), ((1,),), ((1, 2), (-3, -4)), ((1, 2.5, 3), (3, 4, 5), (1, -5.5, 6))]
+ILLEGAL_TYPE_MATRIX_LIST = [None, 1, "test", [(1,)]]
+ILLEGAL_TYPE_MATRIX_CELL = [(("1",),), ((1, 1), (1, [2])), ((1, 1), (1, "[2]"))]
+ILLEGAL_VALUE_MATRIX_LIST = [((), (1,)), ((1,), (1,)), ((1, 2), (1,))]
+VERY_BIG_MATRIX = (((1, ) * 100), ) * 100
+
+
+@pytest.fixture
+def matrix_factory():
+    def _matrix_factory(matrix):
+        return Matrix(matrix)
+
+    return _matrix_factory
+
+
+@pytest.fixture
+def compare_matrix_rows():
+    def _compare_matrix_rows(matrix_a, matrix_b):
+        for row_a, row_b in zip(matrix_a, matrix_b):
+            assert row_a == row_b
+
+    return _compare_matrix_rows
+
+
+########################################################################################################################
+@pytest.mark.parametrize('expected_matrix, expected_exception',
+                         list(zip(LEGAL_MATRIX_LIST, [None] * len(LEGAL_MATRIX_LIST))) +
+                         list(zip(ILLEGAL_TYPE_MATRIX_LIST, [TypeError] * len(ILLEGAL_TYPE_MATRIX_LIST))) +
+                         list(zip(ILLEGAL_TYPE_MATRIX_CELL, [TypeError] * len(ILLEGAL_TYPE_MATRIX_CELL))) +
+                         list(zip(ILLEGAL_VALUE_MATRIX_LIST, [ValueError] * len(ILLEGAL_VALUE_MATRIX_LIST))))
+def test_matrix(matrix_factory, expected_matrix, expected_exception):
```

Comment:
The idea is good generally but it is missing the explecity of testing.
When you are using general functions for test is can means that
1. Maybe not all the tests are needed, for example you can have a hunderds of type that doesn't ok for your matrix and I don't see any reason to test strig and list explecitly.
2. The tests aren't defined well, well defined tests should have indicative names. For example make a dedicated test for test_invalid_elements and use parameterize to use it with diffrent inputs.